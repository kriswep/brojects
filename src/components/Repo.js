import React from 'react'

const Repo = ({ repos = [], currentRepo, onChange }) => (
  <section>
    <select 
        value={currentRepo}        
        onChange={onChange}
        >
      {repos.map((repo) => {
        return (
          <option value={repo.id} key={repo.id}>{repo.full_name}</option>
        );
      })}
    </select>
  </section>
)

export default Repo;
