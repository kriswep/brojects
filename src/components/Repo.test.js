import React from 'react';
import ReactDOM from 'react-dom';
import ReactTestRenderer from 'react-test-renderer';
// import { shallow } from 'enzyme';

import Repo from './Repo';

describe('Repo', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Repo />, div);
  });

  it('renders as expected', () => {
    const repos = [{
      id: 'repoID1',
      full_name: 'repo/1'
    },{
      id: 'repoID2',
      full_name: 'repo/2'
    },
    ]
    const renderer = ReactTestRenderer.create(<Repo repos={repos} currentRepo={'repoID2'}/>);
    // const auth = shallowRenderer.render(<Auth />);

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});