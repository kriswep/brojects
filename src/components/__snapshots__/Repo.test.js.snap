exports[`Repo renders as expected 1`] = `
<section>
  <select
    onChange={undefined}
    value="repoID2">
    <option
      value="repoID1">
      repo/1
    </option>
    <option
      value="repoID2">
      repo/2
    </option>
  </select>
</section>
`;
